### ISSUE TITLE
Problem with the toggle of the Sidebar when zooming

### ISSUE THREAD

**Issue by cedricvoinnet** [coreui/coreui#336]
There is a problem with zooming that makes the height of the sidebar element not fall precisely on the same size as the browser window, so the sidebar is considered not visible and Coreui tries to show it rather than hide it.

https://user-images.githubusercontent.com/10754693/151584063-56315cc3-b59c-453b-a64f-7ebe0fe213de.mp4

Here I'm on the Coreui demo zoomed to 90% and clicking on the sidebar toggler doesn't hide it, the sidebar element is 1195.5556640625px high while the window is 1195px. The zooming causes the pixel count to fall to a decimal point and exceed the size of the window so the Coreui sidebar element is considered not visible.

- All operating system
- All browser

To override this problem, I replaced the method _isVisible of the Sidebar component to make it use Math.floor() to get rid of the decimals. Something like that : 
```js
coreui.Sidebar.prototype._isVisible = sidebarIsVisible

function sidebarIsVisible () {
  const rect = this._element.getBoundingClientRect()
  return rect.top >= 0 &&
    rect.left >= 0 &&
    Math.floor(rect.bottom) <= (window.innerHeight || document.documentElement.clientHeight) &&
    Math.floor(rect.right) <= (window.innerWidth || document.documentElement.clientWidth)
}
```

**Comment by mrholek** [comment]
@cedricvoinnet thank you for your help, I will add this fix in the next release.

**Comment by mrholek** [comment]
Fixed in v4.1.1
